src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use std::io;
+
+pub struct Config {
+    pub version: String,
+}
+
+impl Config {
+    pub fn new() -> Config {
+        let version = format!(
+            "{} {}",
+            env!("CARGO_PKG_NAME").to_string(),
+            env!("CARGO_PKG_VERSION").to_string()
+        );
+
+        Config { version }
+    }
+}
+
+pub fn run(config: Config, stdout: &mut io::Write) {
+    let version_string = format!("{}\n", config.version);
+
+    stdout
+        .write(version_string.as_bytes())
+        .expect("Something went wrong.");
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    struct StdoutDouble {
+        pub written_content: String,
+    }
+
+    impl StdoutDouble {
+        fn new() -> StdoutDouble {
+            let version = "";
+
+            StdoutDouble {
+                written_content: version.to_string(),
```

Comment:
I think `written_content: String::new()` or `written_content: "".into()` should work as well?